flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/tracker/SplitGraphInconsistencyTracker.java

```diff
@@ -53,16 +53,22 @@ public void addNodes(List<Shard> shards) {
         for (Shard shard : shards) {
             addNode(shard);
         }
+        for (Shard shard : shards) {
```

Comment:
Let's add a comment here why we need 2x loops

Reply:
makes sense, updated